autowebcompat/utils.py

```diff
@@ -226,3 +228,24 @@ def read_bounding_boxes(file_name):
 def write_bounding_boxes(bounding_boxes, file_name):
     with open(file_name, 'w') as f:
         print(json.dumps(bounding_boxes), file=f)
+
+
+def get_machine_info():
+    lscpu = subprocess.check_output('lscpu', shell=True).strip().decode()
+    lscpu = lscpu.split('\n')
+    parameter_value_map = {}
+    for row in lscpu:
+        row = row.split(':')
+        parameter_value_map[row[0]] = row[1].strip()
+    return parameter_value_map
+
+
+def write_train_info(information, file_name=None):
+    if file_name is None:
+        file_name = subprocess.check_output('uname -n', shell=True).strip().decode()
+        file_name += datetime.now().strftime('_%H_%M_%Y_%m_%d.txt')
```

Comment:
The machine info is unlikely to change for the same user, so perhaps we should just use the user name.

Reply:
For the filename? If a user tries different models one after the other then filenames should be different.

Reply:
If we store other information and not only the machine info, yes.

Reply:
I didn't realize this was still a WIP, sorry.

Reply:
@marco-c Sorry I forgot to mark it as WIP.
Presently I am storing 
network, optimizer, classification_type, early_stopping, Accuracy, GPU1, GPU2, CPU(s), Thread(s) per core, Core(s) per socket, Model name.
Should I add anything or remove something?

Reply:
Sounds good, we can add more if we thing of something.

Reply:
We can add time taken to train with number of training samples and testing time with number of samples also.

Reply:
@marco-c I was thinking to add model summary to the file. Is it a good idea? We are already saving the name of the model though!

Reply:
@marco-c I have added the model summary part. Since we were returning a model from the function, I have implemented a recursive approach so that even if we join multiple models in future we can get their summary.